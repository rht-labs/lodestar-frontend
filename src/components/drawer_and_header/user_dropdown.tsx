import React, { useState } from 'react';
import { useSession } from '../../context/auth_context/auth_context';
import { Dropdown, DropdownItem, DropdownToggle } from '@patternfly/react-core';
import { Link } from 'react-router-dom';
import { UserProfile } from '../../schemas/user_profile';

export interface UserDropdown {}

export function UserDropdown(props: UserDropdown) {
  const authContext = useSession();
  const [isOpen, setIsOpen] = useState(false);
  const dropdownItems = [
    <DropdownItem value="logOut" key="logout" component={Link} to="/logout">
      Log Out
    </DropdownItem>,
  ];

  const onSelect = (_?: React.SyntheticEvent<HTMLDivElement>) => {
    setIsOpen(false);
  };

  return (
    <div
      style={{
        width: 'auto',
        height: 50,
        justifyContent: 'center',
        display: 'flex',
        alignItems: 'center',
      }}
    >
      <Dropdown
        isPlain={true}
        dropdownItems={dropdownItems}
        toggle={
          <DropdownToggle onToggle={() => setIsOpen(!isOpen)}>
            {UserProfile.getDisplayName(authContext?.sessionData?.profile)}
          </DropdownToggle>
        }
        isOpen={isOpen}
        onSelect={onSelect}
      />
    </div>
  );
}
