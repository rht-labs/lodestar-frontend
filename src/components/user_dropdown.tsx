import React, { useContext, useState } from 'react';
import { SessionContext } from '../context/session_context';
import { Dropdown, DropdownToggle, DropdownItem } from '@patternfly/react-core';
import { Link } from 'react-router-dom';
export interface UserDropdown {}
export function UserDropdown(props: UserDropdown) {
  const sessionContext = useContext(SessionContext);
  const [isOpen, setIsOpen] = useState(false);
  const dropdownItems = [
    <DropdownItem value="logOut" key="logout">
      <Link to="/logout">Log Out</Link>
    </DropdownItem>,
  ];

  const onSelect = (_: React.SyntheticEvent<HTMLDivElement>) => {
    setIsOpen(false);
  };

  return (
    <div
      style={{
        width: 'auto',
        height: 50,
        justifyContent: 'center',
        display: 'flex',
        alignItems: 'center',
      }}
    >
      <Dropdown
        isPlain={true}
        dropdownItems={dropdownItems}
        toggle={
          <DropdownToggle onToggle={() => setIsOpen(!isOpen)}>
            Hello, {sessionContext.profile?.displayName}
          </DropdownToggle>
        }
        isOpen={isOpen}
        onSelect={onSelect}
      ></Dropdown>
    </div>
  );
}
