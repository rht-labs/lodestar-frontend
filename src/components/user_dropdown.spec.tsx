import React from 'react';
import { UserDropdown } from './user_dropdown';

describe('User Dropdown UI', () => {
  test('should match snapshot', () => {
    expect(<UserDropdown />).toMatchSnapshot();
  });
});
