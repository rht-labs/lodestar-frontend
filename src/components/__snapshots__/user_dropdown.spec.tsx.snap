// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Dropdown UI should match snapshot 1`] = `<UserDropdown />`;
